pkg/stanza/operator/helper/splitter.go

```diff
@@ -0,0 +1,44 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helper // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+
+import (
+	"bufio"
+)
+
+// Splitter consolidates Flusher and dependent splitFunc
+type Splitter struct {
+	Encoding  Encoding
+	SplitFunc bufio.SplitFunc
+	Flusher   *Flusher
+}
```

Comment:
I wonder if we really need this struct. The encoding and flusher are not used in this file. Should splitter factory just return a split func?

Reply:
I think that the splitter factory should only return a split func, so that the responsibilities of the splitter factory are more single.
But if we remove `Splitter`, here we need to modify `readFactory` and `Reader`. Do we need to do this?